pkg/controller/datavolume/util.go

```diff
@@ -297,6 +300,21 @@ func renderPvcSpecVolumeSize(client client.Client, pvcSpec *v1.PersistentVolumeC
 	if err != nil {
 		return err
 	}
+
+	if scName := pvcSpec.StorageClassName; scName != nil {
+		storageClass := &storagev1.StorageClass{}
+		if err := client.Get(context.TODO(), types.NamespacedName{Name: *scName}, storageClass); err == nil {
+			if val, exists := storageClass.Annotations[AnnMinimumSupportedPVCSize]; exists {
+				minSize, err := resource.ParseQuantity(val)
+				if err == nil && requestedSize.Cmp(minSize) == -1 {
```

Comment:
I think it would be helpful to log or return an error if parsing the annotation fails, similar to what was done in the [kubevirt PR](https://github.com/kubevirt/kubevirt/pull/13800/files#diff-2013a813861b9274b3272f633523779e9b73b6db8b21ea2806c8de05d847ac26R441). This way users can be notified if the annotation is set incorrectly.